### PR TITLE
Update Documenting.md

### PR DESCRIPTION
**Pull request by n1313**
This PR updates documentation to fix some typos and (hopefully) better demonstrate an alternative way of writing code examples (the "real" React way), as discussed in #324.

### CODE REVIEW

File: docs/Documenting.md [hunk 1]
```diff
@@ -240,33 +240,37 @@ initialState = { isOpen: false }
 </div>
 ```
 
-You _can_ create `React.Component`s in your code examples:
+Using `initialState`, `state` and `setState` helpers to write examples is easy and convenient, but you can write "real" React code as well. For example, the above example could be written like this:
```
Comment (sapegin) [inline docs/Documenting.md]: I think this paragraph was better. Any JSX code is real React code and the point here that you can create new components inside examples.

File: docs/Documenting.md [hunk 2]
```diff
@@ -240,33 +240,37 @@ initialState = { isOpen: false }
...
+class ModalExample extends React.Component {
   constructor() {
     super()
     this.state = {
-      /* ... */
+      isOpen: false
     }
   }
+  toggleOpen() {
+    this.setState({
```
Comment (sapegin) [inline docs/Documenting.md]: Let's not teach people bad practices: use a functional form of setState ;-)

File: docs/Documenting.md [hunk 3]
```diff
@@ -240,33 +240,37 @@ initialState = { isOpen: false }
...
-    const sortedRows = require('sortabular').sorter({
-      /* ... */
-    })(rows)
     return (
-      <TableProvider columns={columns}>
-        <Table.Header />
-        <Table.Body rows={sortedRows} rowKey="id" />
-      </TableProvider>
+      <div>
+        <button onClick={this.toggleOpen.bind(this)}>Open</button>
```
Comment (sapegin) [inline docs/Documenting.md]: If we bind in the constructor  (or use arrow function if Buble supports it) we could make these lines shorter and easier to read.

File: docs/Documenting.md (hunk 1 above)
Comment (n1313) [reply docs/Documenting.md]: By saying "real React code" I mean "code that can be copy-pasted without modifications into any other React application". How should I rephrase it?

File: docs/Documenting.md (hunk 3 above)
Comment (n1313) [reply docs/Documenting.md]: Sure, I'll move the binding to the constructor.

File: docs/Documenting.md (hunk 2 above)
Comment (n1313) [reply docs/Documenting.md]: I want to keep this example as simple as possible. Functions in setState are not widely used yet, as far as I understand.

Comment (sapegin) [reply docs/Documenting.md]: There are or they should ;-) There are recommended by React docs: https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous

**Comment by n1313** [comment]
Alright, I’ll change that.

File: docs/Documenting.md (hunk 1 above)
Comment (sapegin) [reply docs/Documenting.md]: Maybe something like this:

`initialState`, `state` and `setState` helpers are good to show components in different states, but to allow users copy-paste code without modifications into their React app you may want to use `React.Component` instead. We can rewrite an example above like this:

**Comment by n1313** [comment]
@sapegin updated the PR.

**Comment by sapegin** [comment]
Thanks, merged!

**Pull request merged**